test/cpp/test_xla_sharding.cpp

```diff
@@ -55,17 +55,17 @@ TEST_F(XLAShardingTest, GetShardIndicesForDevices) {
   });
   auto sharding = xla::HloSharding::Tile(mesh).ToProto();
   auto shard_shape = ShardingUtil::GetShardShape(tensor, sharding);
-  auto shard_indices =
-      ShardingUtil::GetShardIndicesForDevices(shard_shape, sharding, devices);
+  auto shard_indices = ShardingUtil::GetShardIndicesForDevices(
+      shard_shape, tensor.sizes().vec(), sharding, devices);
   EXPECT_EQ(shard_indices.size(), devices.size());
   /* Tiled indices should be:
                  dim=0 dim=1
        device=0  [0:4,  0:4]
-       device=1  [0:4,  4:8]
+       device=1  [0:4,  4:7]
        device=2  [4:8,  0:4]
-       device=3  [4:8,  4:8] */
+       device=3  [4:8,  4:7] */
```

Comment:
Shouldn't all 8 be replaced by 7?

Reply:
Only in the second dimension - the underlying tensor's shape is `(8, 7)`. The first dimension wasn't padded, so it remains unchanged.

Reply:
Missed the hidden lines...